packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -0,0 +1,783 @@
+        return textDirection == TextDirection.ltr ? a.rect.left.compareTo(b.rect.left) : -a.rect.right.compareTo(b.rect.right);
+      }
+
+      int compareTopSide(_SortData a, _SortData b) {
+        return a.rect.top.compareTo(b.rect.top);
+      }
+
+      // Get the topmost
+      candidates.sort(compareTopSide);
+      final _SortData topmost = candidates.first;
+      // If there are any others in the band of the topmost, then pick the
+      // leftmost one.
+      final List<_SortData> inBandOfTop = inBand(topmost, candidates).toList();
+      inBandOfTop.sort(compareBeginningSide);
+      if (inBandOfTop.isNotEmpty) {
+        return inBandOfTop.first;
+      }
+      return topmost;
+    }
+
+    final List<_SortData> data = <_SortData>[];
+    for (FocusNode node in nodes) {
+      data.add(_SortData(node));
+    }
+
+    // Pick the initial widget as the one that is leftmost in the band of the
+    // topmost, or the topmost, if there are no others in its band.
+    final List<_SortData> sortedList = <_SortData>[];
+    final List<_SortData> unplaced = data.toList();
+    _SortData current = pickFirst(unplaced);
+    sortedList.add(current);
+    unplaced.remove(current);
+
+    while (unplaced.isNotEmpty) {
+      final _SortData next = pickFirst(unplaced);
+      current = next;
+      sortedList.add(current);
+      unplaced.remove(current);
+    }
+    return sortedList.map((_SortData item) => item.node);
+  }
+
+  // Moves the focus forward or backward in reading order, depending on the
+  // value of the forward argument.
+  bool _move(FocusNode currentNode, {@required bool forward}) {
+    final FocusScopeNode nearestScope = currentNode.nearestScope;
+    nearestScope.policyData = null;
+    final FocusNode focusedChild = nearestScope.focusedChild;
+    if (focusedChild == null) {
+      findFirstFocus(currentNode).requestFocusFromPolicy();
+      return true;
+    }
+    final List<FocusNode> sortedNodes = _sortByGeometry(nearestScope).toList();
+    if (forward && focusedChild == sortedNodes.last) {
+      sortedNodes.first.requestFocusFromPolicy();
+      return true;
+    }
+    if (!forward && focusedChild == sortedNodes.first) {
+      sortedNodes.last.requestFocusFromPolicy();
+      return true;
+    }
+
+    final Iterable<FocusNode> maybeFlipped = forward ? sortedNodes : sortedNodes.reversed;
+    FocusNode previousNode;
+    for (FocusNode node in maybeFlipped) {
+      if (previousNode == focusedChild) {
+        node.requestFocusFromPolicy();
+        return true;
+      }
+      previousNode = node;
+    }
+    return false;
+  }
+
+  @override
+  bool next(FocusNode currentNode) => _move(currentNode, forward: true);
+
+  @override
+  bool previous(FocusNode currentNode) => _move(currentNode, forward: false);
+}
+
+/// A widget that describes an inherited focus policy for focus traversal.
+///
+/// By default, traverses in widget order using
+/// [WidgetOrderFocusTraversalPolicy].
```

Comment:
Wouldn't reading order make a better default?

Reply:
Oh, actually, that is what the default is.  Fixed.